src/server/lib/okta/api/errors.ts

```diff
@@ -0,0 +1,89 @@
+import type { Response } from 'node-fetch';
+import {
+  ErrorResponse,
+  InvalidEmailFormatError,
+  OktaAPIResponseParsingError,
+  OktaError,
+  ResourceAlreadyExistsError,
+  ApiValidationError,
+  ErrorCause,
+  MissingRequiredFieldError,
+  ResourceNotFoundError,
+  ActivateUserFailedError,
+  OperationForbiddenError,
+} from '@/server/models/okta/Error';
+
+export const handleErrorResponse = async (response: Response) => {
+  const error = await extractErrorResponse(response);
+  if (error.errorId && error.errorSummary) {
+    const { errorCode, errorSummary } = error;
+    switch (errorCode) {
+      case 'E0000001': {
+        if (error.errorCauses) {
+          const { errorCauses } = error;
+          if (
+            errorCauses.some((cause) =>
```

Comment:
wondering if it might help readability here if this was abstracted into a method called `errorCausePresent` or something similar because it's repeated a lot?

Reply:
I have refactored this into a function